odoo/addons/base/models/ir_actions.py

```diff
@@ -498,25 +503,29 @@ def log(message, level="info"):
         eval_context = super(IrActionsServer, self)._get_eval_context(action=action)
         model_name = action.model_id.sudo().model
         model = self.env[model_name]
-        record = None
-        records = None
-        if self._context.get('active_model') == model_name and self._context.get('active_id'):
-            record = model.browse(self._context['active_id'])
-        if self._context.get('active_model') == model_name and self._context.get('active_ids'):
-            records = model.browse(self._context['active_ids'])
+        records = model
+
+        if self._context.get('active_model') == model_name:
+            if self._context.get('active_id'):
+                records = model.browse(self._context['active_id'])
+            if self._context.get('active_ids'):
+                records = model.browse(self._context['active_ids'])
         if self._context.get('onchange_self'):
-            record = self._context['onchange_self']
+            records |= self._context['onchange_self']
+
         eval_context.update({
             # orm
             'env': self.env,
             'model': model,
             # Exceptions
             'Warning': odoo.exceptions.Warning,
             # record
-            'record': record,
-            'records': records,
+            'record': records, # backwards compatibility
+            'records': records, # backwards compatibility
+            'self': records,
```

Comment:
Not sure that putting PDB for real in the eval context is a good idea, we just add it locally. Anyway we cannot debug from a saas server. It doesn't make much sense. Maybe just keep the line in comment.

Reply:
you're right, updated :^)